bundles/model/org.eclipse.smarthome.model.core/src/main/java/org/eclipse/smarthome/model/core/internal/ModelRepositoryImpl.java

```diff
@@ -202,6 +218,41 @@ private Resource getResource(String name) {
         return resourceSet.getResource(URI.createURI(name), false);
     }
 
+    private String validateModel(String name, InputStream inputStream) throws IOException {
```

Comment:
Could you add a JavaDoc that explains the return value and the thrown exception (warnings are logged, errors are returned, ...)?